src/main/java/se/iths/springbootgroupproject/controllers/WebController.java

```diff
@@ -53,30 +54,49 @@ public String messagePage(@RequestParam(value = "page", defaultValue = "0") Stri
-            The path should probably change name to /users/profiles and access the oauth2User or /users/profiles/{uniqueValue} and fetch
-            uniqueValue from userService.
-         */
+    @GetMapping("/myprofile")
+    public String userProfile (@RequestParam(value = "page", defaultValue = "0") String page, Model model, @AuthenticationPrincipal OAuth2User principal, HttpServletRequest httpServletRequest) {
+        int p = Integer.parseInt(page);
+        if (p < 0) p = 0;
+        User user = userService.findByGitHubId(principal.getAttribute("id"));
+        List<MessageAndUsername> messages = messageService.findAllMessagesByUser(user, PageRequest.of(p,10));
+        int allMessageCount = messageService.findAllMessagesByUser(user).size();
 
-        System.out.println("User LoginName " + principal.getAttribute("login"));
 
-        User user = userRepository.findByUserName(principal.getAttribute("login")).get();
-        System.out.println("USER FIRSTNAME " + user.getFirstName());
+        model.addAttribute("messages", messages);
+        model.addAttribute("currentPage", p);
+        model.addAttribute("totalMessages", allMessageCount);
+        model.addAttribute("httpServletRequest", httpServletRequest);
+        model.addAttribute("name" , user.getFirstName() + " " + user.getLastName());
+        model.addAttribute("userName", user.getUserName());
+        model.addAttribute("profilepic",user.getImage());
+        model.addAttribute("email", user.getEmail());
 
-        String userName = principal.getAttribute("login");
-        String profilePic = principal.getAttribute("avatar_url");
-        String name = principal.getAttribute("name");
+        return "userprofile";
+    }
 
-        System.out.println(principal.getAttribute("avatar_url").toString());
-        model.addAttribute("name" , name);
-        model.addAttribute("userName", userName);
-        model.addAttribute("profilepic",profilePic);
+    @GetMapping("/myprofile/edit")
+    public String editUserProfile(Model model){
+        model.addAttribute("formData", new EditUserFormData());
+        return "edituser";
+    }
 
-        //Every attribute you can get from github
-        model.addAttribute("all", principal);
-        return "userprofiledemo";
+    @PostMapping("/myprofile/edit")
+    public String editUserProfile(@Valid @ModelAttribute("formData") EditUserFormData userForm,
+                    BindingResult bindingResult,
+                    @AuthenticationPrincipal OAuth2User principal) {
+        if(bindingResult.hasErrors()) {
+            return "edituser";
+        }
+        User user = userService.findByGitHubId(principal.getAttribute("id"));
+        user.setUserName(userForm.getUserName());
+        user.setFirstName(userForm.getFirstName());
+        user.setLastName(userForm.getLastName());
+        user.setEmail(userForm.getEmail());
+        userService.save(user);
+        return "redirect:/web/myprofile";
```

Comment:
I think that this is good enough right now.  It works as it should, it saves the information to the database. We could maybe fix so that the fields in the form are already filled with the current information so that you won't have to fill in the exact same information? Or that you only have to fill the actual fields that you want to change, so that blank fields does not update anything?
Implementing a way to update single fields is not a critera for the project and should be implemented secondarily, after we have filled all critera for the project to pass. 🙂

Reply:
Prefilled fields is a good idea 👍